src/server_manager/web_app/cloud_accounts.ts

```diff
@@ -12,28 +12,129 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import * as cloud from '../model/cloud';
 import * as digitalocean from '../model/digitalocean';
+import * as gcp from '../model/gcp';
+import {DigitalOceanAccount, ShadowboxSettings} from './digitalocean_account';
+import {GcpAccount} from './gcp_account';
 
-type DigitalOceanAccountFactory = (accessToken: string) => digitalocean.Account;
+export type AccountJson = {
+  digitalocean?: DigitalOceanAccountJson,
+  gcp?: GcpAccountJson,
+};
 
-export class CloudAccounts {
-  private readonly DIGITALOCEAN_TOKEN_STORAGE_KEY = 'LastDOToken';
+type DigitalOceanAccountJson = {
+  accessToken: string
+};
+
+type GcpAccountJson = {
+  refreshToken: string,
+};
+
+/**
+ * Manages connected cloud provider accounts.
+ */
+export class CloudAccounts implements cloud.CloudAccounts {
+  private readonly LEGACY_DIGITALOCEAN_STORAGE_KEY = 'LastDOToken';
+  private readonly ACCOUNTS_STORAGE_KEY = 'accounts-storage';
+
+  private digitalOceanAccount: DigitalOceanAccount = null;
+  private gcpAccount: GcpAccount = null;
 
   constructor(
-      private digitalOceanAccountFactory: DigitalOceanAccountFactory,
-      private storage = localStorage) { }
+      private shadowboxSettings: ShadowboxSettings,
+      private isDebugMode: boolean,
+      private storage = localStorage) {}
+
+  /**
+   * Loads the saved cloud accounts from disk.
+   *
+   * NOTE: This method must be called before calls to any other CloudAccounts
+   * methods. Failure to do so may result in accounts being disconnected or
+   * overwritten.
+   */
+  load(): void {
+    const accountJsonsString = this.storage.getItem(this.ACCOUNTS_STORAGE_KEY);
+
+    // Migrate any legacy DigitalOcean access token.
+    if (!accountJsonsString) {
+      const digitalOceanToken = this.storage.getItem(this.LEGACY_DIGITALOCEAN_STORAGE_KEY);
```

Comment:
Can you move this to a loadLegacy() method?
Also, let's return because it's a complete operation and not call save because it's easier to reason about load if it's not changing storage.

Reply:
I'm not sure I understand your suggestion. Are you suggesting that we don't persist the token to the new 'accounts' key?

My intention was to do both parts of the migration (load the existing one and persist it under the new storage key).